packages/js/src/utils/typescript/ts-config.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  joinPathFragments,
+  offsetFromRoot,
+  Tree,
+  updateJson,
+  workspaceRoot,
+} from '@nrwl/devkit';
+import { existsSync } from 'fs';
+import { dirname, join } from 'path';
+
+let tsModule: typeof import('typescript');
+
+export function readTsConfig(tsConfigPath: string) {
+  if (!tsModule) {
+    tsModule = require('typescript');
+  }
+  const readResult = tsModule.readConfigFile(
+    tsConfigPath,
+    tsModule.sys.readFile
+  );
+  return tsModule.parseJsonConfigFileContent(
+    readResult.config,
+    tsModule.sys,
+    dirname(tsConfigPath)
+  );
+}
+
+export function getRootTsConfigPathInTree(tree: Tree): string | null {
+  for (const path of ['tsconfig.base.json', 'tsconfig.json']) {
+    if (tree.exists(path)) {
+      return path;
+    }
+  }
+
+  return 'tsconfig.base.json';
+}
+
+export function getRelativePathToRootTsConfig(
+  tree: Tree,
+  targetPath: string
+): string {
+  return offsetFromRoot(targetPath) + getRootTsConfigPathInTree(tree);
+}
+
+export function getRootTsConfigFileName(): string | null {
+  for (const tsConfigName of ['tsconfig.base.json', 'tsconfig.json']) {
+    const tsConfigPath = join(workspaceRoot, tsConfigName);
+    if (existsSync(tsConfigPath)) {
+      return tsConfigName;
+    }
+  }
+
+  return null;
+}
+
+export function getRootTsConfigPath(): string | null {
+  const tsConfigFileName = getRootTsConfigFileName();
+
+  return tsConfigFileName ? join(workspaceRoot, tsConfigFileName) : null;
+}
+
+export function updateRootTsConfig(
+  host: Tree,
+  options: {
+    name: string;
+    importPath?: string;
+    projectRoot: string;
+    js?: boolean;
+  }
+) {
+  updateJson(host, getRootTsConfigPathInTree(host), (json) => {
+    if (!options.importPath) {
+      throw new Error(
+        `Unable to update ${options.name} using the import path "${options.importPath}". Make sure to specify a valid import path one.`
+      );
+    }
```

Comment:
nit (non-blocking): could be before updateJson callback, since it doesn't need the json value. but doesn't actually matter. just a though :)

Reply:
changed.